lib/msal-common/src/authority/AuthorityFactory.ts

```diff
@@ -17,44 +17,71 @@ import { AdfsAuthority } from "./AdfsAuthority";
-            return AuthorityType.B2C;
+        // This is temporary. Remove when ADFS is supported for browser
+        if (adfsDisabled && acquireTokenAuthority.authorityType == AuthorityType.Adfs) {
+            throw ClientAuthError.createInvalidAuthorityTypeError(acquireTokenAuthority.canonicalAuthority);
+        }
 
-        // defaults to Aad
-        return AuthorityType.Aad;
+        if (acquireTokenAuthority.discoveryComplete()) {
+            return acquireTokenAuthority;
+        }
+
+        try {
+            await acquireTokenAuthority.resolveEndpointsAsync();
+            return acquireTokenAuthority;
+        } catch (e) {
+            throw ClientAuthError.createEndpointDiscoveryIncompleteError(e);
+        }
     }
 
     /**
      * Create an authority object of the correct type based on the url
-     * Performs basic authority validation - checks to see if the authority is of a valid type (eg aad, b2c)
+     * Performs basic authority validation - checks to see if the authority is of a valid type (i.e. aad, b2c, adfs)
+     * 
+     * Does not perform endpoint discovery.
+     * 
+     * @param authorityUrl 
+     * @param networkInterface 
      */
-    public static createInstance(authorityUrl: string, networkInterface: INetworkModule): Authority {
+    static createInstance(authorityUrl: string, networkInterface: INetworkModule): Authority {
         // Throw error if authority url is empty
         if (StringUtils.isEmpty(authorityUrl)) {
             throw ClientConfigurationError.createUrlEmptyError();
         }
 
-        const type = AuthorityFactory.detectAuthorityFromUrl(authorityUrl);
-
-        // Depending on above detection, create the right type.
-        switch (type) {
-            case AuthorityType.Aad:
-                return new AadAuthority(authorityUrl, networkInterface);
-            case AuthorityType.B2C:
-                return new B2cAuthority(authorityUrl, networkInterface);
-            case AuthorityType.Adfs:
-                return new AdfsAuthority(authorityUrl, networkInterface);
-            default:
-                throw ClientAuthError.createInvalidAuthorityTypeError(`${authorityUrl}`);
-        }
+        return AuthorityFactory.detectAuthorityFromUrl(authorityUrl, networkInterface);
+    }
+
+    /**
+     * Parse the url and determine the type of authority.
+     * @param authorityString 
+     * @param networkInterface 
+     */
+    private static detectAuthorityFromUrl(authorityString: string, networkInterface: INetworkModule): Authority {
+        const authorityUrl: UrlString = new UrlString(authorityString);
+        const components = authorityUrl.getUrlComponents();
+        const pathSegments = components.PathSegments;
+
+        if (pathSegments.length && pathSegments[0].toLowerCase() === Constants.ADFS)
+            return new AdfsAuthority(authorityString, networkInterface);
+        else if (B2cAuthority.B2CTrustedHostList.length)
```

Comment:
Can you add brackets please (this will be enforced by lint soon)?

Reply:
done.